cl_dll/hud_strafeguide.cpp

```diff
@@ -0,0 +1,157 @@
+#include <complex>
+
+#include "hud.h"
+#include "cl_util.h"
+
+#include "pm_defs.h"
+#include "pm_movevars.h"
+
+enum border {
+	RED_GREEN,
+	GREEN_WHITE,
+	WHITE_GREEN,
+	GREEN_RED
+};
+
+const float pi = std::acos(-1.);
+
+int CHudStrafeGuide::Init()
+{
+	m_iFlags = HUD_ACTIVE;
+
+	hud_strafeguide = CVAR_CREATE("hud_strafeguide", "0", FCVAR_ARCHIVE);
+	hud_strafeguide_fov = CVAR_CREATE("hud_strafeguide_fov", "140", FCVAR_ARCHIVE);
+	hud_strafeguide_height = CVAR_CREATE("hud_strafeguide_height", "0", FCVAR_ARCHIVE);
+	hud_strafeguide_size = CVAR_CREATE("hud_strafeguide_size", "0", FCVAR_ARCHIVE);
```

Comment:
What does this do? Pretty confusing to have both height and size for a full-screen bar.

Reply:
`hud_strafeguide_height` refers to the vertical position to be consistent with `hud_speedometer_height` and `hud_jumpspeed_height`. That's why the naming is so awkward, `hud_strafeguide_size` controls the fatness of the bar.
At first I made it always use the font size for the size of the bar. That looks good at high (4K) resolution, but at lower resolutions I think it's way too big.